cloudinit/net/eni.py

```diff
@@ -401,6 +401,10 @@ def _render_iface(self, iface, render_hwaddress=False):
         sections = []
         subnets = iface.get('subnets', {})
         accept_ra = iface.pop('accept-ra', None)
+        ethernet_wol = iface.pop('wakeonlan', None)
+        if ethernet_wol is not None:
```

Comment:
If network config with `wakeonlan: false` is specified, then I think we'll enable WOL here: the code in `network_state` will convert `"false"` to `False`, and `False is not None` so we'll still set it here.

Reply:
Yes a silly code mistake on my part and I'd only tested myself with "wakeonlan: true" and no "wakeonlan" setting present and so had missed this.

Fixed now and I've added extra "false" testcases for this setting.